client/src/components/addSourceWizard/addSourceWizardStepTwo.js

```diff
@@ -75,19 +75,31 @@ class AddSourceWizardStepTwo extends React.Component {
       remappedCredentials = credentials.map(val => val.id);
     }
 
+    let udpatedSource;
+    if (nextProps.source !== _.get(this.state, 'source')) {
```

Comment:
There may be some limitations around this check, think using lodash's `_.isEqual(a, b)` might be better

Reply:
I'll update to use _.isEqual for now. I think we need a deeper dive on the use of props and state for this component.